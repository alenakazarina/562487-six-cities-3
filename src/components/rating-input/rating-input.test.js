import React from 'react';
import renderer from 'react-test-renderer';
import RatingInput from './rating-input';

const mockFn = () => {};
const ratings = [
  {
    value: 5,
    title: `perfect`
  },
  {
    value: 4,
    title: `good`
  }
];

describe(`RatingInput`, () => {
  it(`should render RatingInput not checked`, () => {
    const tree = renderer.create(
        <RatingInput
          rating={ratings[0]}
          isChecked={false}
          onChange={mockFn}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it(`should render RatingInput checked`, () => {
    const tree = renderer.create(
        <RatingInput
          rating={ratings[1]}
          isChecked
          onChange={mockFn}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
