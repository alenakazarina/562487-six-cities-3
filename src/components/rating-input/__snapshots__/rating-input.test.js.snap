// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingInput should render RatingInput checked 1`] = `
Array [
  <input
    checked={true}
    className="form__rating-input visually-hidden"
    id="4-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={4}
  />,
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="4-stars"
    title="good"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>,
]
`;

exports[`RatingInput should render RatingInput not checked 1`] = `
Array [
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    id="5-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={5}
  />,
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="5-stars"
    title="perfect"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>,
]
`;
